File: ios/Classes/FltsPlugin.swift
import Flutter
import UIKit
import LTSSDK

public class FltsPlugin: NSObject, FlutterPlugin {
  static var lts:LTSSDK?
    
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flts", binaryMessenger: registrar.messenger())
    let instance = FltsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initLts":
        initLts(call, result: result)
    case "setLocalLogLevel":
        setLocalLogLevel(call, result: result)
    case "report":
        report(call, result: result)
    case "reportImmediately":
        reportImmediately(call, result: result)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
    private func setLocalLogLevel(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if let levelName = call.arguments as? String  {
            var level = LTSLoggerLevel.off;
            if (levelName == "DEBUG") {
                level = .debug
            }else if(levelName == "ERROR") {
                level = .error
            }else if(levelName == "INFO") {
                level = .info
            }else if(levelName == "OFF") {
                level = .off
            }else if(levelName == "WARNING") {
                level = .warn
            }
            LTSSDK.setLogLevel(level)
            result(true)
        }else {
            result(false)
        }
    }
    
    private func initLts(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if let arguments = call.arguments as? [String: Any]  {
            // LTS参数配置
            let params = LTSConfigParams()
            // 必填参数
            params.region = arguments["region"] as? String ?? ""
            params.projectId = arguments["projectId"] as? String ?? ""
            params.groupId = arguments["groupId"] as? String ?? ""
            params.streamId = arguments["streamId"] as? String ?? ""
            params.accessKey = arguments["accessKey"] as? String ?? ""
            params.secretKey = arguments["secretKey"] as? String ?? ""
            // 选填参数
//            params.url = "https://lts-access.cn-north-4.myhuaweicloud.com" //要上报的LTS公网地址域名
//            params.cacheThreshold = 200
//            params.timeInterval = 3
//            params.isReportWhenEnterBackgroundEnabled = true
//            params.isReportWhenAPPLaunchEnabled = false

            // 初始化打印日志级别（默认.off）
            LTSSDK.setLogLevel(.off)
            // LTS初始化方法
            let lts = LTSSDK(config:params)
            FltsPlugin.lts = lts
            print("初始化LTSSDK 成功：\(params.region)")
            result(true)
        }else {
            result(false)
        }
    }
    
    /// 上报
    private func report(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if let lts = FltsPlugin.lts {
            if let arguments = call.arguments as? [String: Any]  {
                if let logs = arguments["logs"] as? [[String: Any]],let labels = arguments["labels"]  as? [String: Any]{
                    lts.report(logs, labels: labels );
                }
            }
        }  else {
            print("""
                       ===================================================
                       LTS SDK 未初始化
                       ===================================================
                      """)
        }
       
    }
    
    private func reportImmediately(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if let lts = FltsPlugin.lts {
            if let arguments = call.arguments as? [String: Any]  {
                if let logs = arguments["logs"] as? [[String: Any]],let labels = arguments["labels"]  as? [String: String]{
                    DispatchQueue(label: "com.kyy.lts").async {
//                        lts.report([["content1_key": "content1_value"], ["content2_key": "content2_value"]], labels: ["label_key": "label_value"])
                        lts.reportImmediately(logs, labels: labels );
                    }
                }
            }
        }  else {
            print("""
                       ===================================================
                       LTS SDK 未初始化
                       ===================================================
                      """)
        }
    }
}

File: android/src/main/kotlin/com/cc/flts/FltsPlugin.kt
package com.cc.flts

import android.app.Application
import androidx.annotation.NonNull
import com.cloud.LTSSDK
import com.cloud.UserConfig
import com.cloud.util.log.LogLevel

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import kotlin.collections.HashMap as HashMap1

/** FltsPlugin */
class FltsPlugin : FlutterPlugin, MethodCallHandler {
    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    var mContext: android.app.Application? = null
    var lts: LTSSDK? = null
    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flts")
        mContext = flutterPluginBinding.applicationContext as android.app.Application;
        channel.setMethodCallHandler(this)
    }


    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        when (call.method) {
            "initLts" -> {
                initLts(call, result)
            }

            "setLocalLogLevel" -> {
                setLocalLogLevel(call, result)
            }

            "report" -> {
                report(call, result)
            }

            "reportImmediately" -> {
                reportImmediately(call, result)
            }

            else -> {
                result.notImplemented()
            }
        }
    }

    private fun initLts(@NonNull call: MethodCall, @NonNull result: Result) {
        var arguments = call.arguments as Map<String, String>;
        val userConfig = UserConfig.Builder().setRegion(arguments["region"] ?: "")  // 云日志服务的区域
            .setProjectId(arguments["projectId"] ?: "")  // 华为云账号的项目ID
            .setGroupId(arguments["groupId"] ?: "")  // LTS的日志组ID
            .setStreamId(arguments["streamId"] ?: "")  // LTS的日志流ID
            .setAccessKey(arguments["accessKey"] ?: "")  // 华为云访问密钥
            .setSecretKey(arguments["secretKey"] ?: "")  // 华为云秘密访问秘钥
            .build()
        var lts = LTSSDK(mContext!!, userConfig)
        if (lts.isInitialized) {
            print("LTSSDK isInitialized:true");
        }
        this.lts = lts
    }

    private fun setLocalLogLevel(@NonNull call: MethodCall, @NonNull result: Result) {
        var levelName = call.arguments as? String ?: return
        var level: kotlin.Int = LogLevel.OFF
        when (levelName) {
            "DEBUG" -> {
                level = LogLevel.DEBUG
            }

            "ERROR" -> {
                level = LogLevel.ERROR
            }

            "INFO" -> {
                level = LogLevel.INFO
            }

            "OFF" -> {
                level = LogLevel.OFF
            }

            "WARNING" -> {
                level = LogLevel.WARNING
            }
        }
        LTSSDK.setLogLevel(level)
        result.success(true)
    }


    private fun report(call: MethodCall, result: Result) {
        val arguments = call.arguments as? Map<String, Any> ?: return
        val logs = arguments["logs"] as? ArrayList<HashMap1<String, Any>> ?: return
        val labels = arguments["labels"] as? HashMap1<String, Any> ?: return

        lts?.report(logs.toTypedArray(), labels)
        result.success(true)
    }

    private fun reportImmediately(call: MethodCall, result: Result) {
        val arguments = call.arguments as? Map<String, Any> ?: return
        val logs = arguments["logs"] as? ArrayList<HashMap1<String, Any>> ?: return
        val labels = arguments["labels"] as? HashMap1<String, Any> ?: return
        lts?.reportImmediately(logs.toTypedArray(), labels)
        result.success(true)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    }


}
